source/view/blocks.js

```diff
@@ -1662,377 +1662,377 @@ Blockly.JavaScript['triangle_operations_dilate_auto_fixed'] = function( block )
-//         .appendField(new Blockly.FieldTextInput('0',
-//         Blockly.FieldTextInput.numberValidator), "BY")
-//         .appendField(")");
-//     this.appendDummyInput()
-//         .setAlign(Blockly.ALIGN_CENTRE)
-//         .appendField("C ⇒")
-//         .appendField("(")
-//         .appendField(new Blockly.FieldTextInput('0',
-//         Blockly.FieldTextInput.numberValidator), "CX")
-//         .appendField(",")
-//         .appendField(new Blockly.FieldTextInput('0',
-//         Blockly.FieldTextInput.numberValidator), "CY")
-//         .appendField(")");
-//     this.setInputsInline(false);
-//     this.setPreviousStatement(true, 'INPUT');
-//     this.setNextStatement(true);
-//     this.setColour(195);
-//     this.setTooltip('');
-//     this.data = currentBlocklyNodeID;
-//   },
-//   onchange: function() {
-//     if (!this.workspace) {
-//       // Block has been deleted.
-//       return;
-//     }
-
-//     //First check if we put the block in the right spot
+Blockly.Blocks['triangle_operations_dilate_two'] = {
+  init: function() {
+    this.appendDummyInput()
+        .setAlign(Blockly.ALIGN_CENTRE)
+        .appendField("Dilate", "OP")
+        .appendField("(")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "OPX")
+        .appendField(",")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "OPY")
+        .appendField(")");
+    this.appendDummyInput()
+        .appendField(new Blockly.FieldImage("http://i.imgur.com/UpsXHeX.png", 150, 20, "*"));
+    this.appendDummyInput()
+        .setAlign(Blockly.ALIGN_CENTRE)
+        .appendField("A ⇒")
+        .appendField("(")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "AX")
+        .appendField(",")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "AY")
+        .appendField(")");
+    this.appendDummyInput()
+        .setAlign(Blockly.ALIGN_CENTRE)
+        .appendField("B ⇒")
+        .appendField("(")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "BX")
+        .appendField(",")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "BY")
+        .appendField(")");
+    this.appendDummyInput()
+        .setAlign(Blockly.ALIGN_CENTRE)
+        .appendField("C ⇒")
+        .appendField("(")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "CX")
+        .appendField(",")
+        .appendField(new Blockly.FieldTextInput('0',
+        Blockly.FieldTextInput.numberValidator), "CY")
+        .appendField(")");
+    this.setInputsInline(false);
+    this.setPreviousStatement(true, 'INPUT');
+    this.setNextStatement(true);
+    this.setColour(195);
+    this.setTooltip('');
```

Comment:
Be sure to check the other blocks for the correct tooltip formatting here when the tooltip is ultimately added.